scripts/APIDiffReport/Sources/APIDiffReport/diffreport.swift

```diff
@@ -0,0 +1,342 @@
+        "source.lang.swift.decl.function.free": "function",
+        "source.lang.swift.decl.var.global": "global var",
+        "source.lang.swift.decl.protocol": "protocol",
+        "source.lang.swift.decl.enumelement": "enum value"
+        ][kind] {
+        return pretty
+    }
+    return kind
+}
+
+func prettyString(forModificationKind kind: String) -> String {
+    switch kind {
+    case "key.swift_declaration": return "Swift declaration"
+    case "key.parsed_declaration": return "Declaration"
+    case "key.doc.declaration": return "Declaration"
+    case "key.typename": return "Declaration"
+    case "key.always_deprecated": return "Deprecation"
+    case "key.deprecation_message": return "Deprecation message"
+    default: return kind
+    }
+}
+
+/** Walk the APINode to the root node. */
+func rootName(forApi api: APINode, apis: ApiNameNodeMap) -> String {
+    let name = api["key.name"] as! String
+    if let parentUsr = api["parent.usr"] as? String, let parentApi = apis[parentUsr] {
+        return rootName(forApi: parentApi, apis: apis)
+    }
+    return name
+}
+
+func prettyName(forApi api: APINode, apis: ApiNameNodeMap) -> String {
+    let name = api["key.name"] as! String
+    if let parentUsr = api["parent.usr"] as? String, let parentApi = apis[parentUsr] {
+        return "`\(name)` in \(prettyName(forApi: parentApi, apis: apis))"
+    }
+    return "`\(name)`"
+}
+
+/** Normalize data contained in an API node json dictionary. */
+func apiNode(from sourceKittenNode: SourceKittenNode) -> APINode {
+    var data = sourceKittenNode
+    data.removeValue(forKey: "key.substructure")
+    for (key, value) in data {
+        data[key] = String(describing: value)
+    }
+    return data
+}
+
+/**
+ Recursively iterate over each sourcekitten node and extract a flattened map of USR identifier to
+ APINode instance.
+ */
+func extractAPINodeMap(from sourceKittenNodes: [SourceKittenNode]) -> ApiNameNodeMap {
+    var map: ApiNameNodeMap = [:]
+    for file in sourceKittenNodes {
+        for (_, information) in file {
+            let substructure = (information as! SourceKittenNode)["key.substructure"] as! [SourceKittenNode]
+            for jsonNode in substructure {
+                map += extractAPINodeMap(from: jsonNode)
+            }
+        }
+    }
+    return map
+}
+
+/**
+ Recursively iterate over a sourcekitten node and extract a flattened map of USR identifier to
+ APINode instance.
+ */
+func extractAPINodeMap(from sourceKittenNode: SourceKittenNode, parentUsr: String? = nil) -> ApiNameNodeMap {
+    var map: ApiNameNodeMap = [:]
+    for (key, value) in sourceKittenNode {
+        switch key {
+        case "key.usr":
+            if let accessibility = sourceKittenNode["key.accessibility"] {
+                if accessibility as! String != "source.lang.swift.accessibility.public" &&
+                    accessibility as! String != "source.lang.swift.accessibility.open" {
```

Comment:
There we control accessibility level of interest. Also, we can filter out undocumented or explicit ':nodoc:' items.